elasticdl/pkg/kernel/capi/kernel_api.cc

```diff
@@ -11,3 +12,33 @@ void SGD(float* grad, float* param, double lr, long long size) {
 
   ep -= lr * eg;
 }
+
+void Adam(float* grad,
```

Comment:
According to [tensorflow doc ](https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Adam,)there is another bool argument `amsgrad`.

Reply:
AMSGRAD is a modification of Adam, which needs to maintain the running means of m and v,  maybe it is better to write another kernel.

Reply:
When amsgrad == true, one more value is required, `vhat` for the maximum of all 2nd moment running avg so far. We can put it in Adam as other logics are similar.